graphql_compiler/schema_transformation/query_plan.py

```diff
@@ -276,3 +296,164 @@ def _get_plan_and_depth_in_dfs_order_helper(query_plan, depth):
             )
         return plan_and_depth_in_dfs_order
     return _get_plan_and_depth_in_dfs_order_helper(query_plan, 0)
+
+
+def execute_query_plan(schema_id_to_execution_func, query_plan_descriptor, query_args):
+    """Execute the given query plan and return the produced results."""
+    result_components_by_plan_id = {}
+
+    stitching_output_names_by_parent_plan_id = dict()
+    for join_descriptor in query_plan_descriptor.output_join_descriptors:
+        parent_plan_id = join_descriptor.child_query_plan.parent_query_plan.plan_id
+        stitching_output_names_by_parent_plan_id.setdefault(parent_plan_id, []).append(
+            join_descriptor.output_names)
+
+    full_query_args = dict(query_args)
+
+    plan_and_depth = _get_plan_and_depth_in_dfs_order(query_plan_descriptor.root_sub_query_plan)
```

Comment:
Does the code below depend on the order being a dfs order? In some very trivial cases we will have to start execution from the leafs. Just making sure we don't lock ourselves from that execution plan with the code structure.

Reply:
One thing we should document better than we currently have is the distinction between the output of `split_query()` and the output of `make_query_plan()` -- I think that would clarify the situation here. `split_query()` tells us where we cross schemas, but does not specify an execution order. `make_query_plan()` specifies an order between the subqueries, and is free to specify any valid order -- and may choose an optimized order if it has access to statistics etc.

Given that, once a query plan is made, it's always executed from the root onward. We may in the future allow different styles of executors (e.g. DFS, BFS, async + parallel across children, etc.), but this executor function is just a simple sync DFS.